torchmultimodal/modules/layers/attention.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import itertools
+from typing import Optional
+
+import numpy as np
+
+import torch
+import torch.nn as nn
+
+from torchmultimodal.utils.common import tensor_slice
+
+
+class BroadcastPositionEmbedding(nn.Module):
+    r"""Spatiotemporal broadcasted positional embeddings.
```

Comment:
I think we should move this to its own modules (say positional_embedding?). Having it in generic attention.py might be confusing to users?
Also is this coming from a standard model implementation (apart from multigen)?

Reply:
this is the standard implementation for gpt models I believe

Reply:
Position embedding is standard for the GPT decoder transformer. For now we can leave all components related to the GPT transformer in one script since MHA, Full/Axial attention classes are potentially specific to MUGEN as well.

This is position embedding for video data which could be extended for any video data outside of MUGEN.